storage/replica.go

```diff
@@ -368,6 +374,8 @@ func (r *Replica) Destroy(origDesc roachpb.RangeDescriptor) error {
 	}
 	// Clear the map.
 	r.mu.pendingCmds = map[storagebase.CmdIDKey]*pendingCmd{}
+	r.mu.internalRaftGroup = nil
+	r.mu.destroyed = roachpb.NewRangeNotFoundError(r.RangeID)
```

Comment:
I think for now go with a generic error so we can spot it easily: `roachpb.NewErrorf("replica was garbage collected")`

It should never bubble up to the user. If so, we want to know about it.